lib/Remotery.h

```diff
@@ -311,6 +316,26 @@ typedef enum rmtSampleFlags
 } rmtSampleFlags;
 
 
+typedef enum rmtSampleType
+{
+    RMT_SampleType_CPU,
+    RMT_SampleType_CUDA,
+    RMT_SampleType_D3D11,
+    RMT_SampleType_OpenGL,
+    RMT_SampleType_Metal,
+    RMT_SampleType_Count,
+} rmtSampleType;
```

Comment:
Given the other public interfaces in the header, I prefixed it with RMT_.

Might want to be upper case, to match style of `rmtError` enum?

Reply:
This is fine. I wanted to remove all error-messages-as-enums, anyway and replace them with 4 or 5 broader error enums that give you optional message retrieval.